### DISCUSSION TITLE
Empty media type in HTTP 200 responses

### DISCUSSION THREAD

**Discussion by dmandalidis** [resteasy/resteasy#4294] [General]
Hi all,

[RESTEASY-3178](https://issues.redhat.com/browse/RESTEASY-3178) introduced an exception whenever an HTTP 200 response is received without a `Content-Type` response header. 

However, based on my understanding, this sort of handling is stricter than the HTTP specification which does not mandate such a header. Although weird, an HTTP 200 response without a `Content-Type` is valid.

I haven't checked the implications, but I think RESTeasy is safe to assume `application/octet-stream` if `Content-Type` header is not present in HTTP 200 responses.

Quoting [RFC9110](https://datatracker.ietf.org/doc/html/rfc9110#name-content-type):

```
A sender that generates a message containing content SHOULD generate a Content-Type header field in that message unless the intended media type of the enclosed representation is unknown to the sender. If a Content-Type header field is not present, the recipient MAY either assume a media type of "application/octet-stream" ([[RFC2046](https://datatracker.ietf.org/doc/html/rfc2046)], [Section 4.5.1](https://www.rfc-editor.org/rfc/rfc2046#section-4.5.1)) or examine the data to determine its type.
```

also [RFC7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1) states that the response can be empty (philosophically, what's the content-type of an empty response? :) ).

Thanks for reading,
Dimitris

cc @jamezp

**Comment by dmandalidis** [comment]
Hi again,

Any feedback on this?

Thanks

**Reply by jamezp** [reply]
Well @dmandalidis, I owe you a huge apology. I asked you to start a discussion and it got lost in my inbox and now it's 2 months later. I'm really sorry about that.

It should only throw an exception if the response code was 200 and there was an attempt to extract content which is `null`. What I mean is if you simply did:
```java
try (Response response = webTarget.request().get()) {
    final int status = response.getStatus();
    // do something
}
```

Then no exception should be thrown. Where it would be thrown is when you attempt to do something like `response.readEntity(String.class)` or if you did something like `webTarget.request().get(String.class)`.

That said, if this is causing issues if you and you have some use-cases I'm definitely open to hearing about them.
